website/pages/docs/internals/limits.mdx

```diff
@@ -0,0 +1,271 @@
+within a single storage entry.
+
+|   | Consul default (512 KiB) | Integrated storage default (1 MiB) |
+|---|--------------------------|------------------------------------|
+| Maximum policy size | 512 KiB | 1 MiB |
+| Maximum number of policies per namespace | no limit | no limit |
+| Maximum number of policies per token | ~14,000 | ~28,000 |
+| Maximum number of policies per entity or group | ~14,000 | ~28,000 |
+
+Each time a token is used, Vault must assemble the collection of
+policies attached to that token, to the entity, to any groups that the
+entity belongs to, and recursively to any parents of that entity.
+Very large numbers of policies are possible, but can cause Vault’s
+response time to increase. You can monitor the
+[`vault.core.fetch_acl_and_token`](/docs/internals/telemetry#core-metrics)
+metric to determine if the time required to assemble an access control list
+is becoming excessive.
+
+### Versioned key-value store (kv-v2 secret engine)
+
+|                                       |          | 
+|---------------------------------------|----------|
+| Number of secrets | no limit, up to available storage capacity |
+| Maximum size of one version of a secret | slightly less than one storage entry (512 KiB or 1024 KiB) |
+| Number of versions of a secret | default 10; configurable per-secret or per-mount |
+| Maximum number of versions (not checked when configured) | at least 24,000 |
+
+Each version of a secret must fit in a single storage entry; the
+key-value pairs are converted to JSON before storage.
+
+Version metadata consumes 21 bytes per version and must fit in a
+single storage entry, separate from the stored data.
+
+### Transit secret engine
+
+The maximum size of a Transit ciphertext or plaintext is limited by Vault's
+maximum request size, as described [below](#request-size).
+
+All archived versions of a single key must fit in a single storage entry.
+This limit depends on the key size.
+
+| Key length | Consul default (512 KiB) | Integrated storage default (1 MiB) |
+|----------|--------------------------|---------------------------------------|
+| aes128-gcm96    |  2008   |  4017   |
+| aes128-gcm96    |  1865   |  3731   |
+| chacha-poly1305 |  1865   |  3731   |
+| ed25519         |  1420   |  2841   |
+| ecdsa-p256      |   817   |  1635   |
+| ecdsa-p384      |   659   |  1318   |
+| ecdsa-p523      |   539   |  1078   |
+| 1024-bit RSA    |   169   |  333    |
+| 2048-bit RSA    |   116   |  233    |
+| 4096-bit RSA    |    89   |  178    |
+
+## Other limits
+
+### Request size
+
+The maximum size of an HTTP request sent to Vault is limited by
+the `maximum_request_size` option in the [listener stanza](/docs/configuration/listener/tcp). It defaults to 32 MiB.  This value, minus the overhead of
+the HTTP request itself, places an upper bound on any Transit operation,
+and on the maximum size of any key-value secrets.
+
+### Request duration
+
+The default maximum duration of a Vault operation is
```

Comment:
Do we want to remind people that they may have implicit client-side timeouts too, e.g. $VAULT_CLIENT_TIMEOUT?

Reply:
Added (I didnt know this was only 60 seconds!)